src/simtools/visualization/simtel_event_plots.py

```diff
@@ -0,0 +1,612 @@
+        dt = 1.0
+    t = np.arange(n_samp) * dt
+
+    fig, ax = plt.subplots(dpi=300)
+    mesh = ax.pcolormesh(t, pix_idx, w_sel, shading="auto", vmax=vmax)
+    cbar = fig.colorbar(mesh, ax=ax)
+    cbar.set_label(R1_SAMPLES_LABEL)
+    et_name = getattr(getattr(event.trigger, "event_type", None), "name", "?")
+    tel = source.subarray.tel[tel_id]
+    tel_label = getattr(tel, "name", f"CT{tel_id}")
+    ax.set_title(f"{tel_label} waveform matrix ({et_name})")
+    ax.set_xlabel(TIME_NS_LABEL)
+    ax.set_ylabel("pixel id")
+    fig.tight_layout()
+    return fig
+
+
+def plot_simtel_step_traces(
+    filename,
+    tel_id: int | None = None,
+    pixel_step: int = 100,
+    max_pixels: int | None = None,
+):
+    """Plot step-style traces for regularly sampled pixels: pix 0, N, 2N, ..."""
+    # pylint:disable=import-outside-toplevel
+    import numpy as np
+    from ctapipe.io import EventSource
+
+    source = EventSource(filename, max_events=None)
+    event = _select_event_by_type(source)
+
+    r1_tel_ids = sorted(getattr(event.r1, "tel", {}).keys())
+    if r1_tel_ids:
+        tel_id = tel_id or r1_tel_ids[0]
+    else:
+        _logger.warning("Event has no R1 data for traces plot")
+        return None
+
+    waveforms = getattr(event.r1.tel.get(tel_id, None), "waveform", None)
+    if waveforms is None:
+        _logger.warning(NO_R1_WAVEFORMS_MSG)
+        return None
+
+    w = np.asarray(waveforms)
+    if w.ndim == 3:
+        w = w[0]
+    n_pix, n_samp = w.shape
+
+    readout = source.subarray.tel[tel_id].camera.readout
+    try:
+        dt = (1 / readout.sampling_rate).to(u.ns).value
+    except (AttributeError, ZeroDivisionError, TypeError):
+        dt = 1.0
+    t = np.arange(n_samp) * dt
+
+    pix_ids = np.arange(0, n_pix, max(1, pixel_step))
+    if max_pixels is not None:
+        pix_ids = pix_ids[:max_pixels]
+
+    fig, ax = plt.subplots(dpi=300)
+    for pid in pix_ids:
+        ax.plot(t, w[int(pid)], label=f"pix {int(pid)}", drawstyle="steps-mid")
+    ax.set_xlabel(TIME_NS_LABEL)
+    ax.set_ylabel(R1_SAMPLES_LABEL)
+    et_name = getattr(getattr(event.trigger, "event_type", None), "name", "?")
+    tel = source.subarray.tel[tel_id]
+    tel_label = getattr(tel, "name", f"CT{tel_id}")
+    ax.set_title(f"{tel_label} step traces ({et_name})")
+    ax.legend(loc="best", fontsize=7, ncol=2)
+    fig.tight_layout()
+    return fig
+
+
+def _detect_peaks(trace, peak_width, signal_mod):
+    """Return indices of peaks using CWT if available, else find_peaks fallback."""
+    import numpy as np  # pylint: disable=import-outside-toplevel
+
+    peaks = []
+    try:
+        if hasattr(signal_mod, "find_peaks_cwt"):
+            peaks = signal_mod.find_peaks_cwt(trace, widths=np.array([peak_width]))
+        if not np.any(peaks):
+            peaks, _ = signal_mod.find_peaks(trace, prominence=np.max(trace) * 0.1)
+    except (ValueError, RuntimeError, TypeError):
```

Comment:
Here and before: it is notable that we need to catch 3-4 different type of exceptions. This is a bit unusual, is there no other simpler/clear way to do this?